src/main/kotlin/ecommerce/auth/BearerAuthorizationExtractor.kt

```diff
@@ -0,0 +1,28 @@
+package ecommerce.auth
+
+import jakarta.servlet.http.HttpServletRequest
+
+class BearerAuthorizationExtractor : AuthorizationExtractor<String> {
+    override fun extract(request: HttpServletRequest): String {
+        val headers = request.getHeaders(AuthorizationExtractor.AUTHORIZATION)
+        while (headers.hasMoreElements()) {
+            val value = headers.nextElement()
+            if (value.lowercase().startsWith(BEARER_TYPE.lowercase())) {
+                var authHeaderValue = value.substring(BEARER_TYPE.length).trim()
+                request.setAttribute(ACCESS_TOKEN_TYPE, value.substring(0, BEARER_TYPE.length).trim())
+                val commaIndex = authHeaderValue.indexOf(',')
+                if (commaIndex > 0) {
+                    authHeaderValue = authHeaderValue.substring(0, commaIndex)
+                }
+                return authHeaderValue
+            }
+        }
```

Comment:
```suggestion
        while (headers.hasMoreElements()) {
            val value = headers.nextElement()
            if (value.lowercase().startsWith(BEARER_TYPE.lowercase())) {
                val authHeaderValue = value.substring(BEARER_TYPE.length).trim()
                request.setAttribute(ACCESS_TOKEN_TYPE, value.substring(0, BEARER_TYPE.length).trim())
                val commaIndex = authHeaderValue.indexOf(',')
                if (commaIndex > 0) {
                    return authHeaderValue.substring(0, commaIndex)
                }
                return authHeaderValue
            }
        }
```

We can prevent `var` by returning the value immediately :)

Reply:
Yeah! I love it. Thanks!!